src/modules/staking/sora/components/ValidatorsList.vue

```diff
@@ -0,0 +1,340 @@
+            @click="toggleSelectValidator(validator)"
+          />
+        </li>
+      </ul>
+    </s-scrollbar>
+    <div class="blackout" />
+  </div>
+</template>
+
+<script lang="ts">
+import { components, mixins } from '@soramitsu/soraneo-wallet-web';
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator';
+
+import { Components } from '@/consts';
+import { lazyComponent } from '@/router';
+
+import { soraStakingLazyComponent } from '../../router';
+import {
+  emptyValidatorsFilter,
+  recommendedValidatorsFilter,
+  SoraStakingComponents,
+  ValidatorsListMode,
+} from '../consts';
+import StakingMixin from '../mixins/StakingMixin';
+import ValidatorsMixin from '../mixins/ValidatorsMixin';
+import { ValidatorsFilter } from '../types';
+
+import type { ValidatorInfoFull } from '@sora-substrate/util/build/staking/types';
+
+function filterValidators(validators: ValidatorInfoFull[], filter: ValidatorsFilter, search = '') {
+  return validators.filter((validator) => {
+    if (filter.hasIdentity && (!validator.identity || !Object.keys(validator.identity.info).length)) {
+      return false;
+    }
+    if (filter.notSlashed && validator.blocked) {
+      return false;
+    }
+    if (filter.notOversubscribed && validator.isOversubscribed) {
+      return false;
+    }
+    if (filter.twoValidatorsPerIdentity && validator.isOversubscribed) {
+      const validatorsWithSameIdentity = validators.filter(
+        (v) => v.identity?.info.display === validator.identity?.info.display
+      );
+      if (validatorsWithSameIdentity.length > 2) {
+        return false;
+      }
+    }
+    const name = validator.identity?.info.display ? validator.identity?.info.display : validator.address;
+    return name.toLowerCase().includes(search.toLowerCase());
+  });
+}
+
+@Component({
+  components: {
+    TokenInput: lazyComponent(Components.TokenInput),
+    InfoLine: components.InfoLine,
+    StakingHeader: soraStakingLazyComponent(SoraStakingComponents.StakingHeader),
+    ValidatorAvatar: soraStakingLazyComponent(SoraStakingComponents.ValidatorAvatar),
+  },
+})
+export default class ValidatorsList extends Mixins(StakingMixin, ValidatorsMixin, mixins.LoadingMixin) {
+  @Prop({ required: true, type: String }) readonly mode!: ValidatorsListMode;
+
+  ValidatorsListMode = ValidatorsListMode;
+
+  search = '';
+
+  get filteredValidators() {
+    const filtered = filterValidators(this.validators, this.validatorsFilter, this.search);
+    switch (this.mode) {
+      case ValidatorsListMode.RECOMMENDED:
+        return filterValidators(this.validators, recommendedValidatorsFilter, '').slice(0, this.maxNominations);
+      case ValidatorsListMode.USER:
+        return filtered.filter((v) => this.stakingInfo?.myValidators.includes(v.address));
+      default:
+        return filtered;
+    }
+  }
+
+  toggleSelectValidator(validator: ValidatorInfoFull) {
+    if (this.mode === ValidatorsListMode.RECOMMENDED) {
```

Comment:
could be moved to getter and used later => isValidatorModeRecommended

Reply:
Done